WalletWasabi.Tests/UnitTests/Wabisabi/CredentialTests.cs

```diff
@@ -13,6 +13,60 @@ namespace WalletWasabi.Tests.UnitTests.WabiSabi
 {
 	public class CredentialTests
 	{
+		[Fact]
+		[Trait("UnitTest", "UnitTest")]
+		public void Splitting()
+		{
+			// Split 10 sats into 1, 1, 1, 1, 6.
+			var numberOfCredentials = 2;
+			using var rnd = new SecureRandom();
+			var sk = new CredentialIssuerSecretKey(rnd);
+			var client = new WabiSabiClient(sk.ComputeCredentialIssuerParameters(), numberOfCredentials, rnd);
+			var issuer = new CredentialIssuer(sk, numberOfCredentials, rnd);
```

Comment:
Yes, that's my understanding too: https://github.com/zkSNACKs/WalletWasabi/pull/5130#discussion_r572691214

Reply:
I would prefer to have two `WabiSabiClient` one for amounts and the other for weight units. After all the WabiSabiClient is a low-level building block that we can use to build more complex things. In fact, it was always expected by me to have the WabiSabiClient complexity hidden by some higher-level client like a `WabiSabiCoinJoinClient`.

Reply:
Sorry, yes @lontivero, that's what I meant. Definitely, the last thing I want to do is to mess around inside the `WabiSabiClient` (other than generalized it to hold a `long` delta instead of a `Money` delta.)